pkg/annotations/gcp.go

```diff
@@ -0,0 +1,141 @@
+	var (
+		ctx             context.Context
+		platform        configv1.PlatformType
+		namespace       string
+		createdServices []string
+	)
+
+	g.BeforeAll(func() {
+		cfg, err := config.GetConfig()
+		o.Expect(err).ToNot(o.HaveOccurred(), "Failed to GetConfig")
+
+		cl, err = client.New(cfg, client.Options{})
+		o.Expect(err).NotTo(o.HaveOccurred(), "Failed to create Kubernetes client for test")
+		komega.SetClient(cl)
+		ctx = framework.GetContext()
+		platform, err = framework.GetPlatform(ctx, cl)
+		fmt.Println("platform is ", platform)
+		o.Expect(err).ToNot(o.HaveOccurred(), "Failed to get platform")
+		if platform != configv1.GCPPlatformType {
+			g.Skip("Skipping GCP E2E tests")
+		}
+
+		namespace = "default"
+	})
+
+	g.AfterAll(func() {
+		for _, svcName := range createdServices {
+			service := &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      svcName,
+					Namespace: namespace,
+				},
+			}
+			_ = cl.Delete(ctx, service)
+		}
+	})
+
+	g.It("should validate annotations including network-tier and IP changes", func() {
+		g.By("Create service without annotations")
+		service := &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-service-annotation-validation",
+				Namespace: namespace,
+			},
+			Spec: corev1.ServiceSpec{
+				Type:     corev1.ServiceTypeLoadBalancer,
+				Selector: map[string]string{"app": "test"},
+				Ports: []corev1.ServicePort{{
+					Port: 80,
+				}},
+			},
+		}
+		o.Expect(cl.Create(ctx, service)).To(o.Succeed())
+		createdServices = append(createdServices, service.Name)
+
+		var lastIngressIP string
+		o.Eventually(func() (string, error) {
+			updatedService := &corev1.Service{}
+			err := cl.Get(ctx, client.ObjectKey{Name: service.Name, Namespace: namespace}, updatedService)
+			if err != nil {
+				return "", err
+			}
+			if len(updatedService.Status.LoadBalancer.Ingress) > 0 {
+				lastIngressIP = updatedService.Status.LoadBalancer.Ingress[0].IP
+				return lastIngressIP, nil
+			}
+
+			return "", nil
+		}, 2*time.Minute, 10*time.Second).ShouldNot(o.BeEmpty(), "LoadBalancer service did not get an external IP")
+
+		for key, values := range annotationsToTest {
+			for _, value := range values {
+				g.By(fmt.Sprintf("Adding annotation: %s=%s", key, value))
+				latestService := &corev1.Service{}
+				o.Expect(cl.Get(ctx, client.ObjectKey{Name: service.Name, Namespace: namespace}, latestService)).To(o.Succeed())
+
+				if latestService.Annotations == nil {
+					latestService.Annotations = make(map[string]string)
+				}
+				latestService.Annotations[key] = value
+
+				if key == "cloud.google.com/network-tier" && value != "Standard" && value != "Premium" {
+					o.Expect(cl.Update(ctx, latestService)).ToNot(o.Succeed(), "The annotation 'cloud.google.com/network-tier', if specified, must be either 'Standard' or 'Premium'")
+					continue
+				}
+
+				o.Expect(cl.Update(ctx, latestService)).To(o.Succeed())
```

Comment:
`o.Expect(cl.Update(ctx, latestService)).To(o.Succeed(), "Failed to update service with annotation")`